sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Projection.scala

```diff
@@ -22,9 +22,11 @@ package org.apache.spark.sql.catalyst.expressions
  * @param expressions a sequence of expressions that determine the value of each column of the
  *                    output row.
  */
-class InterpretedProjection(expressions: Seq[Expression]) extends Projection {
-  def this(expressions: Seq[Expression], inputSchema: Seq[Attribute]) =
-    this(expressions.map(BindReferences.bindReference(_, inputSchema)))
+class InterpretedProjection(expressions: Seq[Expression], mutableRow: Boolean = false)
+  extends Projection {
+  def this(expressions: Seq[Expression],
+           inputSchema: Seq[Attribute], mutableRow: Boolean = false) =
```

Comment:
Could you explain this `mutableRow` change?  If we do need to keep it, the style here isn't quite right; see https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide for how to properly wrap long parameter lists.